crypto/threads_win.c

```diff
@@ -133,6 +133,11 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
     return 1;
 }
 
+int CRYPTO_atomic_read(int *val, int *ret, CRYPTO_RWLOCK *lock)
+{
+    return CRYPTO_atomic_add(val, 0, ret, lock);
+}
+
```

Comment:
I'm not an expert in this matter, just googled it, but wouldn't this implementation here be more natural?

```C
int CRYPTO_atomic_read(int *val, int *ret, CRYPTO_RWLOCK *lock)
{
    *ret = InterlockedCompareExchange(val, 0, 0);
    return 1;
}
```
See https://msdn.microsoft.com/en-us/library/ms686355(VS.85).aspx. Unless OpenSSL still supports Windows XP, even`InterlockedCompareExchangeAcquire()` could be used.

Reply:
I know nothing about Windows programming :)
I'll switch this over.